File: Game.java
import java.util.Random;
import java.util.Scanner;

public class Game {

    public static void main(String[] args) {

        Random rand = new Random();
        Scanner scanner = new Scanner(System.in);

        System.out.println("welcome to the number guessing game!");

        int randomNumber = rand.nextInt(100);
        int input = 0;
        int guess = -1;
        String[] difficulty = { "easy", "medium", "hard" };
        int[] tries = { 10, 5, 3 };

        System.out.println("please select the difficulty level:");

        while (input != 1 && input != 2 && input != 3) {

            System.out.println(
                    "1. easy (10 guesses) \n2. medium (5 guesses) \n3. hard (3 chances) \n\nenter your choice: ");

            try {
                input = scanner.nextInt();
            } catch (Exception e) {
                System.out.println("please only write an integer between 1 and 3");
                scanner.next();
            }
            if(input > 3){
                System.out.println("please only write an integer between 1 and 3");
            }
        }

        System.out.println("you selected " + difficulty[input - 1] + " difficulty \n");
        while (tries[input - 1] >= 0) {
            if (guess == randomNumber) {
                System.out.println(
                        "congratulations! you guessed the correct answer with " + tries[input - 1] + " guesses left");
                break;
            }

            System.out.println("please enter your guess: ");

            try {
                guess = scanner.nextInt();
            } catch (Exception e) {
                System.out.println("please only write an integers");
                tries[input - 1]++;
                scanner.next();
            }
            tries[input - 1]--;

            if (guess == randomNumber) {
                System.out.println(
                        "congratulations! you guessed the correct answer with " + tries[input - 1] + " guesses left");
                break;
            } else {
                if (guess > randomNumber) {
                    System.out.println("incorrect, my number is smaller");
                } else {
                    System.out.println("incorrect, my number is larger");
                }
            }
        }
        if (guess != randomNumber) {
            System.out.println("sorry you didnt guess the correct answer, was it was: " + randomNumber);

        }

        scanner.close();
    }
}

File: Game.py
import random

def main():
    random_number = random.randint(0,100)
    tries = [10, 5, 3]
    difficulty = ["easy", "medium", "hard"]

    print("welcome to the number guessing game!")

    input_user = 0
    guess = -1

    print("please select the difficulty:")
    
    while input_user not in [1, 2, 3]:
        print("1. Easy (10 guesses)\n2. Medium (5 guesses)\n3. Hard (3 chances)\n\nEnter your choice: ")

        try:
            input_user = int(input())
        except ValueError:
            print("please only write an integer between 1 and 3")
            continue
        
        if input_user not in [1,2,3]:
            print("please only write an integer between 1 and 3")

    print(f"you selected {difficulty[input_user - 1]} difficulty\n")

    attempts = tries[input_user - 1]

    while attempts > 0:
        if guess == random_number:
            print(f"congratulations! you guessed the correct answer with {attempts} guesses left")
            break
        
        print("please enter your guess: ")
        attempts -= 1
        
        try:
            guess = int(input())
        except ValueError:
            print("please only write integers")
            continue

        if guess == random_number:
            print(f"Congratulations! You guessed the correct answer with {attempts} guesses left")
            break
        else:
            if guess > random_number:
                print(f"Incorrect, my number is smaller. {attempts} guesses left.")
            else:
                print(f"Incorrect, my number is larger. {attempts} guesses left.")

    if guess != random_number:
        print(f"sorry you didnt guess the correct answer, was it was: {random_number}")

if __name__ == "__main__":
    main()